src/main/java/com/google/sps/servlets/ReasonServlet.java

```diff
@@ -1,27 +1,65 @@
 package com.google.sps.servlets;
 
+import static com.google.cloud.spanner.TransactionRunner.TransactionCallable;
+
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.ReadOnlyTransaction;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.cloud.spanner.TransactionContext;
+import com.google.gson.Gson;
 import java.io.IOException;
 import java.time.Instant;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that sends reason for use to /reason. (will change to write to database)*/
+/** Servlet that writes Audit to database*/
 @WebServlet("/reason")
 public final class ReasonServlet extends HttpServlet {
+
+  DatabaseClient dbClient;
+  String queryString;
+  String selectedDatabase;
+  private Constants constants = new Constants();
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get the input from the form.
+    response.setContentType(constants.TEXT_TYPE);
+    String reason = request.getParameter("reason");
+    String account = request.getParameter("account");
+    String query = request.getParameter("query");
+    queryString = request.getParameter("queryString");
+    selectedDatabase = "example-db";
+    this.dbClient = DatabaseConnector.getInstance().getDbClient(selectedDatabase);
+    insertUsingDml(this.dbClient,reason,account,query,response);
```

Comment:
I like your use of the helper method. I don't think you technically need to pass in dbClient since it's an instance variable and could be accessed by all methods in the class.

Reply:
That's a great point I missed that, I fixed this now to have less parameters!